types/dompurify/index.d.ts

```diff
@@ -83,13 +83,13 @@ declare namespace DOMPurify {
 
     interface SanitizeElementHookEvent {
         tagName: string;
-        allowedTags: string[];
+        allowedTags: { [key: string]: boolean };
     }
 
     interface SanitizeAttributeHookEvent {
         attrName: string;
         attrValue: string;
         keepAttr: boolean;
-        allowedAttributes: string[];
+        allowedAttributes: { [key: string]: boolean };
```

Comment:
same, as you checked test option, can we amend the test that covers that hook?

```ts
//test the 'SanitizeAttributeHookEvent' type is publicly accessible.
dompurify.addHook('uponSanitizeAttribute', (currentNode: Element, event: DOMPurify.SanitizeAttributeHookEvent) => {
    if (event.attrName && event.attrName.match(/^\w+-\w+$/) && !event.allowedAttributes[event.attrName]) {
        event.allowedAttributes[event.attrName] = true;
    }
});
```
ref:
https://github.com/cure53/DOMPurify/blob/da4663394f59cfbd313996b858cf499689be76c3/test/test-suite.js#L289-L300
Thanks!

Reply:
Done.